### PR TITLE
fix(docs): clarify Otel mappings

### PR DESCRIPTION
**Pull request by wantsui**
I just learned from @mabdinur that the `name` option maps to resource aka resource.name/resource_name so this is a quick doc fix.

Changes:
- name maps to resource
- trace state maps to meta["tracestate"]

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/opentelemetry/__init__.py [hunk 1]
```diff
@@ -73,13 +73,13 @@ def some_function():
       - ``spanID``
       -
     * - ``trace_state``
-      - ``meta["_sampling_priority_v1"]``, ``meta["_dd.origin"]``, ...
+      - ``meta["tracestate"]["_sampling_priority_v1"]``, ``meta["_dd.origin"]``, ...
```
Comment (mabdinur) [inline ddtrace/opentelemetry/__init__.py:76]:
```suggestion
      - ``meta["tracestate"]``, ...
```

**Review by mabdinur — COMMENTED** [review]
LGTM

File: ddtrace/opentelemetry/__init__.py (hunk 1 above)
Comment (wantsui) [inline ddtrace/opentelemetry/__init__.py:76]: @mabdinur would `meta["_dd.origin"]` also be `meta["tracestate"]["_dd.origin"]` ?

Comment (mabdinur) [reply ddtrace/opentelemetry/__init__.py:76]: Oh no, these are separate tags. meta["tracestate"] is superset of  meta[`_sampling_priority_v1`], meta[`_dd.origin`], meta[`_dd.p.dm`] and tracestate values from third party vendors. 

We can simplify this entry to:

```suggestion
      - ``meta["tracestate"]`` 
```

File: ddtrace/opentelemetry/__init__.py [hunk 2]
```diff
@@ -73,13 +73,13 @@ def some_function():
       - ``spanID``
       -
     * - ``trace_state``
-      - ``meta["_sampling_priority_v1"]``, ``meta["_dd.origin"]``, ...
+      - ``meta["tracestate"]``
```
Comment (mabdinur) [inline ddtrace/opentelemetry/__init__.py:76]:
```suggestion
      - ``meta["tracestate"]``
```

**Review by mabdinur — APPROVED** [review]

**Pull request merged**